src/indexes/createIndexes.ts

```diff
@@ -46,15 +46,22 @@ export function createIndexes<T extends IndexableObj>(
     Deleter<T>
   >(isNotUndefined);
 
-  const captureObject: Captor<T> = (obj) => {
+  const captureObject: Captor<T> = (obj: T): T & Extension<T> => {
+
+    const extendedObject = Object.assign(obj, {
+      deleteFromIndex() {
+        deleters.forEach((deleteFromIndexFunc) => deleteFromIndexFunc(obj));
+      },
+      getTarget() {
+        const {getTarget, deleteFromIndex, ...rest} = extendedObject;
+        return rest;
+      }
```

Comment:
🤔 I think we should be adding these methods to the proxy, I think we're better off proxying the original object rather than the extended object

Reply:
Yeah I did wonder that myself but didn't want to change it too much. Are you suggesting we extend the proxy or implement a get trap so that we can invoke these functions from the proxy? 

I know the more traps we add, the slower proxies get.